pkg/reconciler/v1alpha1/service/service_test.go

```diff
@@ -116,6 +116,33 @@ func TestReconcile(t *testing.T) {
 			Eventf(corev1.EventTypeNormal, "Created", "Created Configuration %q", "pinned2"),
 			Eventf(corev1.EventTypeNormal, "Created", "Created Route %q", "pinned2"),
 		},
+	}, {
+		Name: "pinned - with ready config and route",
+		Objects: []runtime.Object{
+			svc("pinned3", "foo", WithReleaseRollout("pinned3-0001"),
+				WithInitSvcConditions),
+			config("pinned3", "foo", WithReleaseRollout("pinned3-0001"), WithGeneration(1),
+				WithLatestCreated, WithObservedGen,
+				// When we see the LatestCreatedRevision become Ready, then we
+				// update the latest ready revision.
+				WithLatestReady),
+			route("pinned3", "foo", WithReleaseRollout("pinned3-0001"),
+				WithDomain, WithDomainInternal, WithAddress, WithInitRouteConditions,
+				WithStatusTraffic(v1alpha1.TrafficTarget{
+					RevisionName: "pinned3-0001",
+					Percent:      100,
+				}), MarkTrafficAssigned, MarkIngressReady),
+		},
+		Key: "foo/pinned3",
+		WantUpdates: []clientgotesting.UpdateActionImpl{{
+			Object: svc("pinned3", "foo", WithReleaseRollout("pinned3-0001"),
+				WithReadyConfig("pinned3-00001"), WithReadyRoute, WithSvcStatusDomain,
```

Comment:
I'd add a comment above this line describing that what's below it are the set of mutations that we expect to occur when an initial Service observes a Ready Route/Configuration.

Reply:
done

Reply:
Sorry, I meant above the specific mutations.

Reply:
I hope that's what you mean ;-)

Reply:
One more take